src/apis.md

```diff
@@ -0,0 +1,165 @@
+# Questions about designing APIs for others
+
+See also the excellent [Rust API guidelines](https://rust-lang.github.io/api-guidelines/about.html).
+The document you're reading aims to provide extra hints which may be especially
+useful to folk coming from C++, but that's the canonical reference.
+
+## When should my type implement `Default`?
+
+Whenever you'd provide a default constructor in C++.
+
+## When should my type implement `From`, `Into` and `TryFrom`?
+
+You should think of these as equivalent to implicit conversions in C++. Just
+as with C++, if there are _multiple_ ways to convert from your thing to another
+thing, don't implement these, but if there's a single obvious conversion, do.
+
+Usually, don't implement `Into` but instead implement `From`.
+
+## How should I expose constructors?
+
+See the previous two answers: where it's simple and obvious, use the standard
+traits to make your object behavior predictable.
+
+If you need to go beyond that, remember you've got a couple of extra toys in Rust:
+
+* A "constructor" could return a `Result<Self>`
+* Your constructors can have names, e.g. `with_capacity`, `pin`
```

Comment:
It might help to give super specific examples, like `Vec::with_capacity`. There's `std::pin`, but it might help people to see a specific type that has a `pin` constuctor.